x/accounts/internal/implementation/implementation.go

```diff
@@ -6,14 +6,23 @@ import (
 
 	"cosmossdk.io/collections"
 	"cosmossdk.io/core/address"
+	"cosmossdk.io/core/gas"
 	"cosmossdk.io/core/header"
+	gogoproto "github.com/cosmos/gogoproto/proto"
+
+	"github.com/cosmos/cosmos-sdk/codec"
 )
 
 // Dependencies are passed to the constructor of a smart account.
 type Dependencies struct {
-	SchemaBuilder *collections.SchemaBuilder
-	AddressCodec  address.Codec
-	HeaderService header.Service
+	SchemaBuilder    *collections.SchemaBuilder
+	AddressCodec     address.Codec
+	HeaderService    header.Service
+	GasService       gas.Service
+	LegacyStateCodec interface {
```

Comment:
what is the reason for legacy here?

Reply:
This expects gogoproto; ideally in the future we add support for protov2 by default